ygen/enumgen.go

```diff
@@ -440,103 +562,132 @@ func findEnumSet(entries map[string]*yang.Entry, compressPaths, noUnderscores, s
 type enumGenState struct {
 	// definedEnums keeps track of generated enum names to avoid conflicts.
 	definedEnums map[string]bool
-	enumSet      *enumSet
+	// enumSet contains the final, deduped generated enumerated value names
+	// to be returned by findEnumSet.
+	enumSet *enumSet
+	// enumeratedTypedefDedupSets stores the set of unique string keys
+	// representing typedef enumeration usages that were mapped to the same
+	// initial generated name using the basic name generation rules.
+	// This is an intermediate storage for these names. While the initial
+	// generated name is ideally stored inside enumSet directly,
+	// enumeration names may collide, and thus need de-duplication before
+	// they're stored inside enumSet. This is the reason for being for all
+	// dedup sets within enumGenState.
+	enumeratedTypedefDedupSets map[string]map[string]bool
+	// enumeratedLeafDedupSets stores the unique string keys representing
+	// enumeration leaves that were mapped to the same initial generated
+	// name using the basic name generation rules.
+	enumeratedLeafDedupSets map[string]map[string]bool
 }
 
 // newEnumGenState creates a new enumGenState instance initialised with the
 // default state required for code generation.
 func newEnumGenState() *enumGenState {
 	return &enumGenState{
-		definedEnums: map[string]bool{},
-		enumSet:      newEnumSet(),
+		definedEnums:               map[string]bool{},
+		enumSet:                    newEnumSet(),
+		enumeratedTypedefDedupSets: map[string]map[string]bool{},
+		enumeratedLeafDedupSets:    map[string]map[string]bool{},
 	}
 }
 
-// enumeratedUnionEntry takes an input YANG union yang.Entry and returns the set of enumerated
-// values that should be generated for the entry. New yang.Entry instances are synthesised within
-// the yangEnums returned such that enumerations can be generated directly from the output of
-// this function in common with enumerations that are not within a union. The name of the enumerated
+// resolveDedupSets scans through all of the dedup sets, and carries out
+// de-duplication between different enum keys that mapped to the same initial
+// generated name. Then it stores these final names in their appropriate name
+// maps within enumSet.
+func (s *enumGenState) resolveDedupSets() {
+	s.resolveDedupSet(s.enumeratedTypedefDedupSets, s.enumSet.uniqueEnumeratedTypedefNames)
```

Comment:
Do these functions need to in-place modify the argument -- is there a reason not to just assign their output to the right variable in `s`? i.e., `s.enumSet.uniqueEnumeratedTypedefNames = s.resolveDedupSet(s.enumeratedTypedefDedupSets)`